### PR TITLE
split request queues in plasma_manager.cc

### PR DESCRIPTION
**Pull request by istoica**

### CODE REVIEW

**Comment by robertnishihara** [comment]
ok to test

File: src/plasma/plasma_manager.cc [hunk 1]
```diff
@@ -670,7 +682,10 @@ void process_data_chunk(event_loop *loop,
                         int events) {
   /* Read the object chunk. */
   ClientConnection *conn = (ClientConnection *) context;
-  PlasmaRequestBuffer *buf = conn->transfer_queue.front();
+  PlasmaRequestBuffer *buf = conn->data_transfer_queue.front(); 
+  assert(buf != NULL);
+  assert(buf->type == MessageType_PlasmaDataReply);
+
```
Comment (elibol) [inline src/plasma/plasma_manager.cc]: Do we need these asserts? Looks like type is set to `MessageType_PlasmaDataReply` in `process_data_request`.

File: src/plasma/plasma_manager.cc [hunk 2]
```diff
@@ -904,14 +921,14 @@ void request_transfer_from(PlasmaManagerState *manager_state,
     transfer_request->type = MessageType_PlasmaDataRequest;
     transfer_request->object_id = fetch_req->object_id;
 
-    if (manager_conn->transfer_queue.size() == 0) {
+    if (manager_conn->data_transfer_queue.size() == 0 && manager_conn->data_transfer_queue.size() == 0) {
```
Comment (fabuzaid21) [inline src/plasma/plasma_manager.cc]: I think the same condition is repeated twice in this `if`; it should probably be
```
 if (manager_conn->data_transfer_queue.size() == 0 && manager_conn->data_request_queue.size() == 0) {
```
instead

Comment (robertnishihara) [reply src/plasma/plasma_manager.cc]: nice catch :)

**Comment by robertnishihara** [comment]
Closing because this will likely be subsumed by the ongoing rewrite of the plasma manager.

**Pull request closed without merging**